includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -562,6 +605,23 @@ public function get_product() {
 		return false;
 	}
 
+	/**
+	 * Returns the login redirect URL.
+	 *
+	 * @param string $redirect Default redirect URL.
+	 * @return string Redirect URL.
+	 */
+	public function get_login_redirect_url( $redirect ) {
+		$url = esc_url_raw( wp_unslash( $_COOKIE['wcpay_payment_request_redirect_url'] ?? '' ) );
```

Comment:
This looked a little bit risky. A bad intentioned user could set this cookie via JS or in the backend and redirect to another site. I've tried changing the value to `https://google.com` in the cookie but it seems there we already have protections to avoid unwanted redirects.

Reply:
> A bad intentioned user could set this cookie via JS or in the backend

In the backend do you mean via a malicious plugin? In that case we would be vulnerable everywhere and I don't think we can avoid it, but it's a different story when it's caused by the website administrators. What we definitely can and always should avoid is attacks from third parties.

Setting the cookie on the front-end or via JS would affect only the user who has set the cookie, not others, unless I create a browser extension to set this cookie to others.

Do you have suggestions on how to implement this differently? If this could get changed to some other URL, what bad things could happen do you foresee?

Reply:
> Do you have suggestions on how to implement this differently? If this could get changed to some other URL, what bad things could happen do you foresee?

it's OK to use this method since I wasn't able to redirect to another domain even when modifying the cookie manually.